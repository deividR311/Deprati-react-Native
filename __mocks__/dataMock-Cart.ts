export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
