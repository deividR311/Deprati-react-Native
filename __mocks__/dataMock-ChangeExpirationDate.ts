export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
