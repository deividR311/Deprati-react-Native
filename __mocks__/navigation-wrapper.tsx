import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
