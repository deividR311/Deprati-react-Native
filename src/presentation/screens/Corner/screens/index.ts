export { default as productosVeganos } from './ProductosVeganos';
export { default as coleccionFrioMujer } from './ColeccionFrioMujer';
export { default as socialmenteConsciente } from './SocialmenteConsciente';
export { default as dior } from './Dior';
