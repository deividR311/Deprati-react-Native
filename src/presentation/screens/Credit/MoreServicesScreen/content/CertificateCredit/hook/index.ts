export { default as useCertificateCredit } from './useCertificateCredit.hook';
