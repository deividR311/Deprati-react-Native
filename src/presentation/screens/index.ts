export * from './Barcode';
export * from './CategoryPage';
export * from './ErrorPage';
//export * from './ProductPage'
export * from './RecoveryPass';
//export * from './Search'
//export * from './SignIn'
export * from './SignUp';
export * from './Splash/SplashScreen';
export * from './Stories';
export * from './Welcome/WelcomeScreen';
export * from './dashboard/PDP/PDPScreen';
export * from './dashboard/PLP/PLPScreen';
export * from './home';
export * from './contactless-payment';
