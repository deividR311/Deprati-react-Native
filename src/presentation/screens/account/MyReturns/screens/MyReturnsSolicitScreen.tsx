//libs
import React, { useCallback, useEffect, useMemo } from 'react';
import { useFocusEffect } from '@react-navigation/native';
import { KeyboardAvoidingView, ScrollView, Text, View } from 'react-native';
import { Divider } from 'react-native-paper';
import RenderHTML from '../../../../common-components/renderHtml/RenderHTML';

//components
import CardOrderReturnClaim from '../components/CardOrderReturnClaim';
import TemplatePage from '../../../../common-components/template-page';
import NoOrders from '../components/NoOrders';
import Pagination from '../../MyOrders/components/Pagination';
import { FontStyles } from '../../../../../application/common/fonts';
import { SkeletonMyReturnSolicit } from '../SkeletonReturns/SkeletonMyReturnSolicit';
import { SearchInput } from '../components/SearchInput';
import { SupportHeaderCell } from '../../support-tickets/components/ticket-section-header';

//hook
import { useMyReturnsSolicit } from '../hooks/useMyReturnsSolicit.hook';
import usePageContent from '../../../../../infrastucture/apis/contentPage/contentPage.hook';

//styles
import { stylesReturnSolicit as styles, tagsStyles } from './stylesMyReturns';

//utils
import layout from '../../../../../application/common/layout';
import { formatDateOrders } from '../../../../../application/utils/formatDate';
import { IComponentsSolicit } from '../interfaces/IMyReturnsSolicit.interface';

const TITLE = 'Selecciona uno de tus pedidos más recientes';
const SUB_TITLE =
  '*Si no encuentras tu número de pedido, comunícate con servicio al cliente.';
const TEXT_BEFORE =
  'Si no encuentras tu compra en el listado, ingresa el número de pedido aquí.';
const TEXT_AFTER =
  'El número del pedido lo encontraras en la factura de compra o en el correo de confirmación.';

const width = layout.window.width;

export function MyReturnsSolicitScreen() {
  const {
    MAX_ORDER,
    getReturnableOrders,
    isLoadingReturnableOrders,
    isErrorReturnableOrders,
    ordersReturns,
    filteredOrdersMade,
    handleMaxPage,
    prevPage,
    nextPage,
    changePage,
    handleGoEnterReturn,
    padding
  } = useMyReturnsSolicit();

  const {
    loading: loadingContent,
    error: errorContent,
    pageContent,
    getDataContent
  } = usePageContent();

  useFocusEffect(
    useCallback(() => {
      getReturnableOrders();
    }, [])
  );

  useEffect(() => {
    getDataContent({
      pageType: 'ContentPage',
      pageLabelOrId: 'myReturnableOrders'
    });
  }, []);

  const components: IComponentsSolicit = useMemo(() => {
    if (pageContent) {
      return pageContent.components;
    }
    return undefined;
  }, [pageContent]);

  if (!ordersReturns || isLoadingReturnableOrders || loadingContent)
    return <SkeletonMyReturnSolicit />;

  if (ordersReturns && ordersReturns.length === 0) return <NoOrders />;

  return (
    <TemplatePage
      loading={isLoadingReturnableOrders || loadingContent}
      skeleton={<SkeletonMyReturnSolicit />}
      isTab={true}
      error={isErrorReturnableOrders}>
      <KeyboardAvoidingView behavior={'position'}>
        <ScrollView
          style={{ marginBottom: 80 }}
          contentContainerStyle={{ paddingBottom: padding }}>
          {components ? (
            <View style={{ marginBottom: 10, paddingHorizontal: 15 }}>
              <RenderHTML
                tagsStyles={tagsStyles}
                contentWidth={width}
                text={
                  components.BottomMyReturnableOrdersParagraphComponent.content
                }
              />
            </View>
          ) : (
            <Text style={[FontStyles.Body_1, styles.title]}>{TITLE}</Text>
          )}
          {/* <Text style={[FontStyles.Body_1, styles.subTitle]}>{SUB_TITLE}</Text> */}
          {filteredOrdersMade().map((x, index) => (
            <View key={`CardOrderReturn-${index}`}>
              {x.month && (
                <SupportHeaderCell title={formatDateOrders(x.placed)} />
              )}
              {!x.month && <Divider style={styles.divider} />}
              <CardOrderReturnClaim
                typeCard="myReturns"
                date={x.placed}
                code={x.code}
                reasonOrItem={x.totalUnitCount}
                price={x.total.formattedValue}
                stateOrder={x.paymentStatusDisplay}
                onSolicit={() => handleGoEnterReturn(x.code)}
              />
            </View>
          ))}
          {ordersReturns.length > MAX_ORDER && (
            <>
              <Pagination
                maxPag={handleMaxPage()}
                onPrev={() => prevPage()}
                onNext={() => nextPage()}
                onPress={x => changePage(x)}
              />
              <Divider style={styles.divider} />
              <View style={{}}>
                <Text style={[FontStyles.Body_1, styles.paragraph]}>
                  {components?.Footer1MyReturnableOrdersParagraphComponent
                    .content ?? TEXT_BEFORE}
                </Text>

                <SearchInput
                  containerStyle={{ marginBottom: 10 }}
                  onSearch={x => handleGoEnterReturn(x)}
                  numeric
                />
                <Text style={[FontStyles.Body_1, styles.paragraph2]}>
                  {components?.Footer2MyReturnableOrdersParagraphComponent
                    .content ?? TEXT_AFTER}
                </Text>
              </View>
            </>
          )}
        </ScrollView>
      </KeyboardAvoidingView>
    </TemplatePage>
  );
}
