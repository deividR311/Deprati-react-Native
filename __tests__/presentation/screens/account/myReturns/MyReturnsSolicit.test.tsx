import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../../__mocks__/navigation-wrapper';
import { DataMock_MyReturnsSolicit } from '../../../../../__mocks__/dataMock-MyReturns';
import { useMyReturnsSolicit } from '../../../../../src/presentation/screens/account/MyReturns/hooks/useMyReturnsSolicit.hook';
import { MyReturnsSolicitScreen } from '../../../../../src/presentation/screens/account/MyReturns/screens/MyReturnsSolicitScreen';

jest.mock(
  '../../../../../src/presentation/screens/account/MyReturns/hooks/useMyReturnsSolicit.hook'
);

describe('MyReturnsSolicitScreen', () => {
  it('renders correctly in Snapshot', () => {
    useMyReturnsSolicit.mockImplementation(() => {
      return {
        getDetailReturn: jest.fn(),
        MAX_ORDER: 5,
        isLoadingReturnableOrders: false,
        isErrorReturnableOrders: false,
        ordersReturns: DataMock_MyReturnsSolicit,
        getReturnableOrders: jest.fn(),
        filteredOrdersMade: jest.fn(() => DataMock_MyReturnsSolicit),
        handleMaxPage: jest.fn(),
        prevPage: jest.fn(),
        nextPage: jest.fn(),
        changePage: jest.fn(),
        handleGoEnterReturn: jest.fn(),
        padding: 0
      };
    });
    const screenTest = render(
      <NavigationWrapper screen={MyReturnsSolicitScreen} />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });
});
