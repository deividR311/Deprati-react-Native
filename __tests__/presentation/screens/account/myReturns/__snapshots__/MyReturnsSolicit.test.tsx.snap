// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyReturnsSolicitScreen renders correctly in Snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            onLayout={[Function]}
          >
            <View
              style={
                Object {
                  "bottom": 0,
                }
              }
            >
              <RCTScrollView
                contentContainerStyle={
                  Object {
                    "paddingBottom": 0,
                  }
                }
                style={
                  Object {
                    "marginBottom": 80,
                  }
                }
              >
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#000000",
                          "fontFamily": "Roboto-Regular",
                          "fontSize": 16,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0.44,
                          "lineHeight": 24,
                        },
                        Object {
                          "color": "#263238",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 24,
                          "marginTop": 10,
                          "paddingHorizontal": 15,
                        },
                      ]
                    }
                  >
                    Selecciona uno de tus pedidos más recientes
                  </Text>
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "rgba(0, 0, 0, 0.12)",
                            "height": 0.5,
                          },
                          undefined,
                          Object {
                            "alignSelf": "center",
                            "height": 1,
                            "width": "95%",
                          },
                        ]
                      }
                    />
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#ffffff",
                            "borderRadius": 4,
                            "elevation": 4,
                            "marginVertical": 11,
                            "paddingTop": 10,
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 2,
                              "width": 0,
                            },
                            "shadowOpacity": 0.25,
                            "shadowRadius": 3.84,
                          },
                          Object {
                            "marginHorizontal": 16,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "height": 130,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "height": "100%",
                              "width": "90%",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "marginHorizontal": 16,
                                "marginVertical": 10,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000000",
                                  "fontFamily": "Roboto-Medium",
                                  "fontSize": 15,
                                  "fontStyle": "normal",
                                  "fontWeight": undefined,
                                  "letterSpacing": 1,
                                }
                              }
                            >
                              No. Pedido: #
                              4616004
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#979797",
                                  "fontFamily": "Roboto-Medium",
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "letterSpacing": 0.8,
                                }
                              }
                            >
                              Fecha: 
                              31/08/2022
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#979797",
                                  "fontFamily": "Roboto-Medium",
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "letterSpacing": 0.8,
                                }
                              }
                            >
                              Estado de pago: 
                              Pendiente
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#979797",
                                  "fontFamily": "Roboto-Medium",
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "letterSpacing": 0.8,
                                }
                              }
                            >
                              Artículos: 
                              9
                            </Text>
                            <Text
                              style={
                                Object {
                                  "color": "#979797",
                                  "fontFamily": "Roboto-Medium",
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "letterSpacing": 0.8,
                                }
                              }
                            >
                              Total: 
                              $82,58
                            </Text>
                          </View>
                        </View>
                      </View>
                      <View
                        accessibilityLabel=""
                        accessibilityState={
                          Object {
                            "disabled": false,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "backgroundColor": "#E31A2C",
                            "borderRadius": 5,
                            "justifyContent": "center",
                            "marginBottom": 15,
                            "opacity": 1,
                            "paddingVertical": 10,
                            "width": "92%",
                          }
                        }
                        testID=""
                      >
                        <View
                          style={
                            Object {
                              "flexDirection": "row",
                              "justifyContent": "space-between",
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "white",
                                  "fontFamily": undefined,
                                  "fontSize": 14,
                                  "fontStyle": "normal",
                                  "fontWeight": undefined,
                                },
                                Object {},
                                Object {
                                  "letterSpacing": 0.75,
                                  "lineHeight": 16,
                                },
                              ]
                            }
                          >
                            SOLICITAR DEVOLUCIÓN
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          />
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
